wagtail_streamfield_migration_toolkit/test/tests/test_migrations.py

```diff
@@ -0,0 +1,353 @@
+            revisions_from=revisions_from,
+        )
+        migration.operations = [migration_operation]
+
+        loader = MigrationLoader(connection=connection)
+        loader.build_graph()
+        project_state = loader.project_state()
+        schema_editor = connection.schema_editor(atomic=migration.atomic)
+        migration.apply(project_state, schema_editor)
+
+
+class TestNonPageModelWithoutRevisions(BaseMigrationTest):
+    model_name = "SampleModel"
+
+    @classmethod
+    def setUpTestData(cls):
+        instances = [None for i in range(3)]
+        instances[0] = factories.SampleModelFactory(
+            content__0__char1="Test char 1",
+            content__1__char1="Test char 2",
+            content__2__char2="Test char 3",
+            content__3__char2="Test char 4",
+        )
+        instances[1] = factories.SampleModelFactory(
+            content__0__char1="Test char 1",
+            content__1__char1="Test char 2",
+            content__2__char2="Test char 3",
+        )
+        instances[2] = factories.SampleModelFactory(
+            content__0__char2="Test char 1",
+            content__1__char2="Test char 2",
+            content__2__char2="Test char 3",
+        )
+
+        cls.instances = {
+            instance.id: instance.content.raw_data for instance in instances
+        }
+
+    def test_migrate_stream_data(self):
+        self.apply_migration()
+
+        instances = models.SampleModel.objects.all().annotate(
+            raw_content=Cast(F("content"), JSONField())
+        )
+
+        for instance in instances:
+            prev_content = self.instances[instance.id]
+            for old_block, new_block in zip(prev_content, instance.raw_content):
+                self.assertEqual(old_block["id"], new_block["id"])
+                if old_block["type"] == "char1":
+                    self.assertEqual(new_block["type"], "renamed1")
+                else:
+                    self.assertEqual(old_block["type"], new_block["type"])
+
+
+class TestPage(BaseMigrationTest):
+    model_name = "SamplePage"
+
+    @classmethod
+    def setUpTestData(cls):
+        instances = [None for i in range(3)]
+        instances[0] = factories.SamplePageFactory(
+            content__0__char1="Test char 1",
+            content__1__char1="Test char 2",
+            content__2__char2="Test char 3",
+            content__3__char2="Test char 4",
+        )
+        instances[1] = factories.SamplePageFactory(
+            content__0__char1="Test char 1",
+            content__1__char1="Test char 2",
+            content__2__char2="Test char 3",
+        )
+        instances[2] = factories.SamplePageFactory(
+            content__0__char2="Test char 1",
+            content__1__char2="Test char 2",
+            content__2__char2="Test char 3",
+        )
+
+        cls.instances = {}
+        cls.revisions = {}
+        for instance in instances:
+            cls.instances[instance.id] = instance.content.raw_data
```

Comment:
Is this used?

Reply:
yeah that is used in one of the test methods to get the data from before applying the migration

```
    for instance in instances:
        prev_content = self.instances[instance.id]
```